### PR TITLE
[Faker] Remove non-existent methods

### PR DESCRIPTION
**Pull request by testica**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/Marak/faker.js/tree/v4.1.0>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by testica** [comment]
Resubmitted https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42255

**NOTE**
The latest npm version for faker is the [4.1.0](https://github.com/Marak/faker.js/tree/v4.1.0).
For some reason master updates have not been published.

Also, you can confirm downloading the npm package.

File: types/faker/faker-tests.ts [hunk 1]
```diff
@@ -92,10 +89,7 @@ resultStr = faker.finance.currencyCode();
 resultStr = faker.finance.currencyName();
 resultStr = faker.finance.currencySymbol();
 resultStr = faker.finance.bitcoinAddress();
-resultStr = faker.finance.creditCardNumber();
-resultStr = faker.finance.creditCardNumber('visa');
 resultStr = faker.finance.creditCardCVV();
```
Comment (mattbishop) [inline types/faker/faker-tests.ts]: I can't find this method (finance.creditCardCVV()) in 4.1.0

File: types/faker/index.d.ts [hunk 2]
```diff
@@ -15,7 +16,6 @@ declare namespace Faker {
 
 		address: {
 			zipCode(format?: string): string;
-			zipCodeByState(state: string): string;
```
Comment (PranavSenthilnathan) [inline types/faker/index.d.ts]: I see this in the source [here](https://github.com/Marak/faker.js/blob/72dce1198fd348f069b8e42e91ef67b74da1009f/lib/address.js#L39). It's not on the readme but it look like it's in the package. Is this also the case with the other methods? I don't see the 4.1.0 package on npm so I can't confirm.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/faker/index.d.ts [hunk 3]
```diff
@@ -3,7 +3,8 @@
 // Definitions by: Ben Swartz <https://github.com/bensw>,
 //                 Bas Pennings <https://github.com/basp>,
 //                 Yuki Kokubun <https://github.com/Kuniwak>,
-//                 Matt Bishop <https://github.com/mattbishop>
+//                 Matt Bishop <https://github.com/mattbishop>,
+//                Leonardo Testa <https://github.com/testica>
```
Comment (mattbishop) [inline types/faker/index.d.ts]: Need one more space before Leonardo

**Review by mattbishop — CHANGES_REQUESTED** [review]
Still need to remove finance.creditCardCVV() or show where it lives.

File: types/faker/index.d.ts (hunk 2 above)
Comment (testica) [reply types/faker/index.d.ts]: You are comparing to master branch, but for [4.1.0](https://github.com/Marak/faker.js/blob/v4.1.0/lib/address.js) you can check that `zipCodeByState` doesn't exist.

**Comment by mattbishop** [comment]
Approved.

**Pull request merged**